### PR TITLE
calibrationEDD can be run from the command line with a config file

### PR DESCRIPTION
**Pull request by loichuder**
I run `calibrationEDD` a first time. For ease of use, it is now possible to run the file from the command line by supplying a YML configuration file.

I did some minor improvements underlined below.

### CODE REVIEW

File: easistrain/EDD/calibrationEDD.py [hunk 1]
```diff
@@ -1,22 +1,12 @@
...
-# sample = 'align'
-# dataset = '0001'
-# scanNumberHorizontalDetector = '66'
-# scanNumberVerticalDetector = '65'
-# nameHorizontalDetector = 'mca2_det0'
-# nameVerticalDetector = 'mca2_det1'
-# numberOfBoxes = 4
-# nbPeaksInBoxes = [1,2,1,1]
-# rangeFit = [620,780,1020,1120,3500,3800,3850,4090]
-# sourceCalibrantFile = '/home/esrf/slim/easistrain/easistrain/EDD/BaSource'
```
Comment (loichuder) [inline easistrain/EDD/calibrationEDD.py:19]: Moved to `example_config/calibrationEDD.yml`

File: easistrain/EDD/calibrationEDD.py [hunk 2]
```diff
@@ -1,22 +1,12 @@
+import argparse
```
Comment (loichuder) [inline easistrain/EDD/calibrationEDD.py:1]: Note that I renamed the file `calibrationEdd` to `calibrationEDD` for consistency with the other files

File: easistrain/EDD/calibrationEDD.py [hunk 3]
```diff
@@ -144,21 +134,22 @@ def guessParameters(xData, yData, counterOfBoxes, nbPeaksInBoxes):
...
+    sample: str,
+    dataset: Union[str, int],
+    scanNumberHorizontalDetector: Union[str, int],
+    scanNumberVerticalDetector: Union[str, int],
+    nameHorizontalDetector: str,
+    nameVerticalDetector: str,
+    numberOfBoxes: int,
+    nbPeaksInBoxes: Sequence[int],
+    rangeFit: Sequence[int],
+    sourceCalibrantFile: str,
```
Comment (loichuder) [inline easistrain/EDD/calibrationEDD.py:149]: I added some type indications. These do not change anything to the code execution but are a nice addition for the developer to know what kind of arguments are expected.

File: easistrain/EDD/calibrationEDD.py [hunk 4]
```diff
@@ -184,7 +175,7 @@ def calibEdd(
         ]  ## calibration pattern of vertical detector
 
     h5Save = h5py.File(fileSave, "a")  ## create h5 file to save in
-    if not "detectorCalibration" in h5Save.keys():
+    if "detectorCalibration" not in h5Save.keys():
```
Comment (loichuder) [inline easistrain/EDD/calibrationEDD.py:178]: Easier to read. Also raised by `flake8`

File: easistrain/EDD/calibrationEDD.py [hunk 5]
```diff
@@ -634,3 +625,13 @@ def calibEdd(
...
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("config_file", type=str, help="Path to the config file")
+    args = parser.parse_args()
+
+    config = read_config_file(args.config_file)
+    calibEdd(**config)
+    print("Program finished !")
```
Comment (loichuder) [inline easistrain/EDD/calibrationEDD.py:637]: The part that allow to run the file with a config file path as arg

File: setup.cfg [hunk 6]
```diff
@@ -36,3 +37,10 @@ dev =
     %(test)s
     black
     flake8
+
+[flake8]
+max-line-length = 88
+# E203, W503 disabled because incompatible with black formatting
+# E262 (inline should begin by #) ignored because of numerous ## comments
+# E501 (line too long) ignored because of the end-of-line comments
+extend-ignore = E203, E262, E501, W503
```
Comment (loichuder) [inline setup.cfg:46]: I added a sensible config for flake8. You can run flake8 to catch issues (such as undefined variable names or invalid formattion) using `flake8 [path_to_the_file]`.

With this config, `calibrationEDD.py` passes the `flake8` check. As I progress in the test writings, all files should pass the check.

**Review by mohamedfaresslim — APPROVED** [review]
Ok for all

**Pull request merged**